api/language-extensions/syntax-highlight-guide.md

```diff
@@ -13,16 +13,27 @@ Syntax highlighting determines the color and style of source code displayed in t
 
 There are two components to syntax highlighting:
 
-- Breaking text into a list of tokens and scopes using a grammar
-- Then using a theme to map these scopes to specific colors and styles
+- [Tokenization](#tokenization): Breaking text into a list of tokens
+- [Theming](#theming): Using themes or user settings to map the tokens to specific colors and styles
 
-This document only discusses the first part: breaking text into tokens and scopes that existing color themes can colorize. For more information about customizing the styling of different scopes in the editor, see the [Color Theme Guide](/api/extension-guides/color-theme#syntax-colors)
+Before diving into the details, a good start is to play with the [scope inspector](#scope-inspector) and explore what tokens are present in a source file and what theme rules they match to. To see both semantic and syntax token, use a built-in theme (e.g. Dark+) on a TypeScript file.
 
-## TextMate grammars
+## Tokenization
 
-VS Code uses [TextMate grammars][tm-grammars] to break text into a list of tokens. TextMate grammars are a structured collection of [Oniguruma regular expressions](https://macromates.com/manual/en/regular_expressions) and are typically written as a plist or JSON. You can find a good introduction to TextMate grammars [here](https://www.apeth.com/nonblog/stories/textmatebundle.html), and you can take a look at existing TextMate grammars to learn more about how they work.
+The tokenization of text is about breaking the text into segments and to classify each segment with a token type.
 
-### Tokens and scopes
+VSCode uses a two step approach:
+
+- [Syntax tokenization](#syntax-tokenization) is based on lexical rules that are expressed as regular expressions contained in a TextMate grammar. The TextMate engine runs in the same process as the renderer and tokens are updated as the user types.
+
+- [Semantic tokenization](#semantic-tokenization) is optional on applied on top of syntax tokens. Semantic tokens come from language servers. A language server has the full understanding of the source file and can classify each symbol identifier with the symbol it resolves to. A constant variable name is rendered as constant throughout the file, not just in its declaration. Same for parameter names, property names, class names and so on.
+As language servers take a while to load and analyze the project, semantic tokens come in with a delay.
+
+## Syntax tokenization
+
+VS Code uses [TextMate grammars][tm-grammars] as the syntax tokenization engine. TextMate grammars are a structured collection of [Oniguruma regular expressions](https://macromates.com/manual/en/regular_expressions) and are typically written as a plist or JSON. You can find a good introduction to TextMate grammars [here](https://www.apeth.com/nonblog/stories/textmatebundle.html), and you can take a look at existing TextMate grammars to learn more about how they work.
+
+### TextMate tokens and scopes
```

Comment:
I don't think making this a very long essay containing two approaches to syntax highlighting is a good idea. TextMate grammar is already confusing enough that I think we should not mix in any semantic highlighting content into this topic.